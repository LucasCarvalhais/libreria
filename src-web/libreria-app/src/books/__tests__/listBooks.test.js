import React from 'react';
import { shallow } from 'enzyme';
import axios from 'axios';
import ListBooks from '../ListBooks';

const books = [{
    bookId: 1,
    title: "Un",
    description: "Uno",
    author: "One",
    edition: 1,
},
{
    bookId: 2,
    title: "Deux",
    description: "Dos",
    author: "Two",
    edition: 1,
}];

describe('ListBooks', () => {
    test('should show book table', () => {
        const component = shallow(<ListBooks />).setState({
            books: books,
            error: null,
        });
        expect(component.find('BookTable')).toHaveLength(1);
    });

    test('should show error message', () => {
        const component = shallow(<ListBooks />).setState({
            books: [],
            error: 'Error',
        });
        expect(component.find('ErrorMessage')).toHaveLength(1);
    });
});