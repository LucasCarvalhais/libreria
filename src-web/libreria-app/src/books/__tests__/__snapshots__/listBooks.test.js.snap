// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`listBooks has a valid snapshot 1`] = `
<div>
  <h1>
    Lista de los libros
  </h1>
  <div>
    <div>
      <p
        className="loading"
      >
        Cargando...
      </p>
    </div>
  </div>
</div>
`;
